static/app/components/core/button/index.tsx

```diff
@@ -110,6 +112,165 @@ interface ButtonPropsWithAriaLabel extends BaseButtonProps {
 
 export type ButtonProps = ButtonPropsWithoutAriaLabel | ButtonPropsWithAriaLabel;
 
+export function Button({
+  size = 'md',
+  disabled,
+  type = 'button',
+  title,
+  tooltipProps,
+  ...props
+}: ButtonProps) {
+  const {handleClick, hasChildren, accessibleLabel} = useButtonFunctionality({
+    ...props,
+    type,
+    disabled,
+  });
+
+  if ('to' in props || 'href' in props) {
+    throw new Error('Button does not support `to` or `href` props');
+  }
```

Comment:
why don’t we keep the types as `never` to avoid having to have a runtime check?

Reply:
This should have not been commited here. The intention was to run the tests in CI to just see if anything breaks. I will remove it.

For button, to or href should not be valid props anyways, so I dont we need the never type.